src/main/java/uk/gov/pay/card/db/loader/BinRangeDataLoader.java

```diff
@@ -0,0 +1,100 @@
+package uk.gov.pay.card.db.loader;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.card.db.CardInformationStore;
+import uk.gov.pay.card.model.CardInformation;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.util.function.Function;
+
+public class BinRangeDataLoader {
+
+    private static final Logger logger = LoggerFactory.getLogger(BinRangeDataLoader.class);
+    private String filePath;
+    private String delimeter;
+    private String dataRowIdentifier;
+    private Function<String[], CardInformation> cardInformationExtractor;
+
+    public static final String WORLDPAY_DELIMITER = "!";
+    public static final String DISCOVER_DELIMITER = ",";
+    public static final String WORLDPAY_ROW_IDENTIFIER = "05";
+    public static final String DISCOVER_ROW_IDENTIFIER = "02";
+
+    public static final Function<String[], CardInformation> WORLDPAY_CARD_INFORMATION_EXTRACTOR = entry -> new CardInformation(
+            entry[4], entry[8], entry[4], Long.valueOf(entry[1]), Long.valueOf(entry[2]));
+
+    public static final Function<String[], CardInformation> DISCOVER_CARD_INFORMATION_EXTRACTOR = entry -> new CardInformation(
+            entry[4], entry[3], entry[4], Long.valueOf(entry[1]), Long.valueOf(entry[2]));
+
+
+    public BinRangeDataLoader(String filePath, String delimeter, String dataRowIdentifier, Function<String[], CardInformation> cardInformationExtractor) {
```

Comment:
Constructor does need to have delimiters? passing as a parameters constants contained within the same class does make much sense. It seems like a factory for BinRangeDataLoaders